src/bridge/ERC20Bridge.sol

```diff
@@ -24,15 +33,27 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 contract ERC20Bridge is AbsBridge, IERC20Bridge {
     using SafeERC20 for IERC20;
 
-    /// @inheritdoc IERC20Bridge
     address public nativeToken;
+    uint8 public nativeTokenDecimals;
 
     /// @inheritdoc IERC20Bridge
     function initialize(IOwnable rollup_, address nativeToken_) external initializer onlyDelegated {
         if (nativeToken_ == address(0)) revert InvalidTokenSet(nativeToken_);
         nativeToken = nativeToken_;
         _activeOutbox = EMPTY_ACTIVEOUTBOX;
         rollup = rollup_;
+
+        // store number of decimals used by native token
+        try ERC20(nativeToken_).decimals() returns (uint8 decimals) {
+            if (decimals > MAX_ALLOWED_NATIVE_TOKEN_DECIMALS) {
+                revert NativeTokenDecimalsTooLarge(decimals);
+            }
+            nativeTokenDecimals = decimals;
+        } catch {
+            // decimal is not part of the ERC20 spec
+            // assume it have 0 decimals if it does not have decimals() method
+            nativeTokenDecimals = 0;
```

Comment:
I feel like we've discussed this before, but if the erc20 has no decimals might it not be better to assume 18 or revert and insist that a decimal value is provided in the constructor here. Do we have examples of tokens that dont use the decimals func? Do they actually use 0 decimals?

Reply:
571303f